examples/99-advanced/ray-trace-moeller.py

```diff
@@ -0,0 +1,144 @@
+"""
+.. _moeller_ray_trace_example:
+
+Visualize the Moeller–Trumbore Algorithm
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example demonstrates the Moeller–Trumbore intersection algorithm
+using pyvista.
+
+For additional details, please reference the following:
+
+- `Möller–Trumbore intersection algorithm <https://en.wikipedia.org/wiki/M%C3%B6ller%E2%80%93Trumbore_intersection_algorithm>`_
+- `Fast Minimum Storage Ray Triangle Intersectio <https://cadxfem.org/inf/Fast%20MinimumStorage%20RayTriangle%20Intersection.pdf>`_
+
+First, define the ray triangle intersection method.
+"""
+
+import pyvista as pv
+import numpy as np
+
+
+def ray_triangle_intersection(ray_start, ray_vec, triangle):
+    """Moeller–Trumbore intersection algorithm.
+
+    Parameters
+    ----------
+    ray_start : np.ndarray
+       Length three numpy array representing start of point.
+
+    ray_vec : np.ndarray
+        Direction of the ray.
+
+    triangle : np.ndarray
+        ``3 x 3`` numpy array containing the three vertices of a
+        triangle.
+
+    Returns
+    -------
+    bool
+        ``True`` when there is an intersection.
+
+    tuple
+        Length three tuple containing the distance ``t``, and the
+        intersection in unit triangle ``u``, ``v`` coordinates.  When
+        there is no intersection, these values will be:
+        ``[np.nan, np.nan, np.nan]``
+
+    """
```

Comment:
This function is cool. How about add this function to `pyvista` library?

Reply:
I agree that it would be cool to have, but I'm not sure about this.  We already have a built-in VTK ray tracing method, and there's even another method that use `pyembree` that's really cool.  This is really for demonstration purposes of an algorithm.

What I'm trying to do is avoid pyvista from becoming too much of a general purpose library and trying to keep in line with @banesullivan's vision of being the 3d `matplotlib` for python.

Plus, this algorithm is pretty slow.  I have a much faster one that I use that needs to be compiled with cython and used kdtree indexing and other magic.  It's more reliable than VTK's method though, which is why I use it.

I'd be fine with making a package for that outside of `pyvista` and adding it to our list of libraries.  Maybe we should do that instead.

Reply:
@akaszynski  Thank you for your comment. I could understand why we should not add functions to the library.